### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by NikitaKarbazin**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,88 @@
...
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+):Student[] {
+  let newSortArrayWithStudents:Student[];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case 'Name':
```
Comment (Olexandr6) [inline src/sortStudents.ts:29]: Let's use SortType here in cases check

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,88 @@
...
+          - nextStudent.grades
+            .reduce((prev, inn) => prev + inn) / nextStudent.grades.length;
+        })
+        : [...students].sort((prevStudent, nextStudent) => {
+          return nextStudent.grades
+            .reduce((prev, inn) => prev + inn) / nextStudent.grades.length
+            - prevStudent.grades
+              .reduce((prev, inn) => prev + inn) / prevStudent.grades.length;
+        });
+      break;
```
Comment (Olexandr6) [inline src/sortStudents.ts:73]: Let's create a separate function to count average grade

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,88 @@
...
+  order: SortOrder,
+):Student[] {
+  let newSortArrayWithStudents:Student[];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case 'Name':
+      newSortArrayWithStudents = order === 'asc'
+        ? [...students]
```
Comment (Olexandr6) [inline src/sortStudents.ts:31]: Let's make a copy of student array once not every time before `sort`

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,88 @@
...
+          .sort((prevStudent, nextStudent) => {
+            return Number(prevStudent.married) - Number(nextStudent.married);
+          })
+        : [...students]
+          .sort((prevStudent, nextStudent) => {
+            return Number(nextStudent.married) - Number(prevStudent.married);
+          });
+  }
```
Comment (Olexandr6) [inline src/sortStudents.ts:85]: Let's better to create a separate case for married, don't put such logic in default case

File: src/sortStudents.ts [hunk 5]
```diff
@@ -1,16 +1,88 @@
...
+
+    case 'Surname':
+      newSortArrayWithStudents = order === 'asc'
+        ? [...students]
+          .sort((prevStudent, nextStudent) => prevStudent.surname
+            .localeCompare(nextStudent.surname))
+        : [...students]
+          .sort((prevStudent, nextStudent) => nextStudent.surname
+            .localeCompare(prevStudent.surname));
+      break;
```
Comment (Olexandr6) [inline src/sortStudents.ts:47]: Look, here you do the same in two cases with `Name` and `Surname`. Only one difference is you sort by a different prop. Let's combine these two cases into one.
For example:
```
case sortBy.Name:
case sortBy.Surname:
    students.sort(studnetA, studentB) => studnetA[sortBy].localCompare(studentB[sortBy])
```

**Review by Olexandr6 — CHANGES_REQUESTED** [review]
Let's work on impovements👇

File: src/sortStudents.ts (hunk 4 above)
Comment (NikitaKarbazin) [reply src/sortStudents.ts:85]: image

**Review by Olexandr6 — APPROVED** [review]

**Pull request closed without merging**